lib/state-string-parser.js

```diff
@@ -1,9 +1,11 @@
-module.exports = function(stateString) {
-	return stateString.split('.').reduce(function(stateNames, latestNameChunk) {
-		if (stateNames.length) {
-			latestNameChunk = stateNames[stateNames.length - 1] + '.' + latestNameChunk
-		}
-		stateNames.push(latestNameChunk)
+module.exports = stateString => {
+	return stateString.split('.').reduce((stateNames, latestNameChunk) => {
+		stateNames.push(
+			stateNames.length
+				? stateNames[stateNames.length - 1] + '.' + latestNameChunk
```

Comment:
```js
`${stateNames[stateNames.length - 1]}.${latestNameChunk}`
```
to be consistent, you could change this, but I think it's more readable the way it is now...
https://github.com/TehShrike/abstract-state-router/blob/aa1a23d6e58ed0210c3b7e41ca5b1dc928858c44/lib/state-state.js#L63

Reply:
Yeah, I think I did change that to a template string, and then ended up changing it back...